src/concurrency/thread.rs

```diff
@@ -628,6 +624,34 @@ impl<'mir, 'tcx: 'mir> ThreadManager<'mir, 'tcx> {
     }
 }
 
+impl VisitMachineValues for ThreadManager<'_, '_> {
+    fn visit_machine_values(&self, visit: &mut impl FnMut(&Operand<Provenance>)) {
+        // FIXME some other fields also contain machine values
```

Comment:
I see there are other fields with machine values, but it's not clear to me that for example `thread_local_alloc_ids` is keeping pointers live. Are there other fields?

Reply:
Isn't that the only way you will reach the current value of thread-local variables? And a thread-local variable can store a pointer to somewhere else.

Reply:
It's similar, I think, to the global base tag map in Stacked Borrows -- I don't know how your GC is not failing left and right due to removing the base tags.

Reply:
`Stack::retain` never removes the bottom-most tag.

Reply:
Ah. Yeah that would help but needs to be carefully documented.